app/src/main/java/org/dhis2/usescases/settings/SyncManagerFragment.java

```diff
@@ -230,14 +233,28 @@ public void showSmsSettings(boolean enabled, String number, boolean waitForRespo
         }
     }
 
+    private void validatePhone(String str) {
+        if (str.startsWith("+")
+            && str.length() > 4) {
+            ((TextInputLayout) binding.settingsSms.findViewById(R.id.settings_sms_receiver_layout)).setError(null);
+        } else if (!str.startsWith("+") && str.length() > 1) {
+            ((TextInputLayout) binding.settingsSms.findViewById(R.id.settings_sms_receiver_layout))
+                    .setError(binding.getRoot().getContext().getResources().getString(R.string.invalid_phone_number));
+            ((SwitchCompat) binding.settingsSms.findViewById(R.id.settings_sms_switch)).setChecked(false);
+        } else {
+            ((TextInputLayout) binding.settingsSms.findViewById(R.id.settings_sms_receiver_layout)).setError(null);
+        }
+        presenter.smsNumberSet(str);
+    }
```

Comment:
can you move logic to the presenter and the call view methods ?
can you add unit tests?